crates/polars-ops/src/frame/join/hash_join/single_keys.rs

```diff
@@ -40,6 +41,7 @@ where
         // Compute the number of elements in each partition for each portion.
         let per_thread_partition_sizes: Vec<Vec<usize>> = keys
             .par_iter()
+            .with_max_len(1)
```

Comment:
Ah, did you see that rayon made larger chunks? Or just to be extra sure that we get 1 partition per thread?

Reply:
Just to be extra sure, since that's what we want here. I was originally doing it to check if it wasn't going wrong, but ultimately I figured might as well leave it in, it can't hurt.